File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/kotlin/shapes.kt
package shapes

// BEGIN PROBE F14.P3
interface Shape {
    fun area(): Double
}

class Circle(val r: Double) : Shape {
    override fun area(): Double = 3.141592653589793 * r * r
}

class Rect(val w: Double, val h: Double) : Shape {
    override fun area(): Double = w * h
}
// END PROBE F14.P3

File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/typescript/shapes.ts
// BEGIN PROBE F14.P3
export interface Shape {
area(): number
}

export class Circle implements Shape {
constructor(private r: number) {}
area(): number {
return 3.141592653589793 * this.r * this.r
}
}

export class Rect implements Shape {
constructor(private w: number, private h: number) {}
area(): number {
return this.w * this.h
}
}
// END PROBE F14.P3
